gio/src/auto/file.rs

```diff
@@ -103,6 +103,158 @@ impl File {
         }
     }
 
+    #[cfg(any(feature = "v2_74", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_74")))]
+    #[doc(alias = "g_file_new_tmp_async")]
+    pub fn new_tmp_async<P: FnOnce(Result<(File, FileIOStream), glib::Error>) + 'static>(
+        tmpl: Option<impl AsRef<std::path::Path>>,
+        io_priority: glib::Priority,
+        cancellable: Option<&impl IsA<Cancellable>>,
+        callback: P,
+    ) {
+        let main_context = glib::MainContext::ref_thread_default();
+        let is_main_context_owner = main_context.is_owner();
+        let has_acquired_main_context = (!is_main_context_owner)
+            .then(|| main_context.acquire().ok())
+            .flatten();
+        assert!(
+            is_main_context_owner || has_acquired_main_context.is_some(),
+            "Async operations only allowed if the thread is owning the MainContext"
+        );
+
+        let user_data: Box_<glib::thread_guard::ThreadGuard<P>> =
+            Box_::new(glib::thread_guard::ThreadGuard::new(callback));
+        unsafe extern "C" fn new_tmp_async_trampoline<
+            P: FnOnce(Result<(File, FileIOStream), glib::Error>) + 'static,
+        >(
+            _source_object: *mut glib::gobject_ffi::GObject,
+            res: *mut crate::ffi::GAsyncResult,
+            user_data: glib::ffi::gpointer,
+        ) {
+            let mut error = ptr::null_mut();
+            let mut iostream = ptr::null_mut();
+            let ret = ffi::g_file_new_tmp_finish(res, &mut iostream, &mut error);
+            let result = if error.is_null() {
+                Ok((from_glib_full(ret), from_glib_full(iostream)))
+            } else {
+                Err(from_glib_full(error))
+            };
+            let callback: Box_<glib::thread_guard::ThreadGuard<P>> =
+                Box_::from_raw(user_data as *mut _);
+            let callback: P = callback.into_inner();
+            callback(result);
+        }
+        let callback = new_tmp_async_trampoline::<P>;
+        unsafe {
+            ffi::g_file_new_tmp_async(
+                tmpl.as_ref().map(|p| p.as_ref()).to_glib_none().0,
+                io_priority.into_glib(),
+                cancellable.map(|p| p.as_ref()).to_glib_none().0,
+                Some(callback),
+                Box_::into_raw(user_data) as *mut _,
+            );
+        }
+    }
+
+    #[cfg(any(feature = "v2_74", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_74")))]
+    pub fn new_tmp_future(
+        tmpl: Option<impl AsRef<std::path::Path>>,
+        io_priority: glib::Priority,
+    ) -> Pin<
+        Box_<dyn std::future::Future<Output = Result<(File, FileIOStream), glib::Error>> + 'static>,
+    > {
+        let tmpl = tmpl.map(|tmpl| tmpl.as_ref().to_owned());
```

Comment:
Don't edit autogenerated files. You'll need to make this one manual, or fix gir.

Reply:
Yea I know, It's easier to get the diff that way to file the issue. I will make it manual in a bit.